core/src/style/polylineStyle.cpp

```diff
@@ -145,15 +145,26 @@ struct PolylineStyleBuilder : public StyleBuilder {
     std::vector<MeshData<V>> m_meshData;
 
     float m_tileUnitsPerMeter;
-    float m_tileSize;
+    float m_tileSizePixels;
     int m_zoom;
 };
 
 template <class V>
-void PolylineStyleBuilder<V>::setup(const Tile& _tile) {
-    m_tileUnitsPerMeter = _tile.getInverseScale();
-    m_zoom = _tile.getID().z;
-    m_tileSize = _tile.getProjection()->TileSize();
+void PolylineStyleBuilder<V>::setup(const Tile& tile) {
+
+	auto id = tile.getID();
+
+    // Use the 'style zoom' to evaluate style parameters.
+    m_zoom = id.s;
+    m_tileUnitsPerMeter = tile.getInverseScale();
+    m_tileSizePixels = tile.getProjection()->TileSize();
+
+    // When a tile is overzoomed, we are actually styling the area of its
+    // 'source' tile, which will have a larger effective pixel size at the
+    // 'style' zoom level.
+    if (id.s > id.z) {
+        m_tileSizePixels *= std::exp2(id.s - id.z);
```

Comment:
It shouldn't be less (save for programmer error), but it can be (and usually is) equal.